src/components/_modal/pages/pickup/PickupSelection.js

```diff
@@ -0,0 +1,277 @@
+import { useRef, useState, useEffect } from "react";
+import List from "@/components/base/forms/list";
+import Text from "@/components/base/text";
+import Translate from "@/components/base/translate";
+// eslint-disable-next-line css-modules/no-unused-class
+import styles from "./Pickup.module.css";
+import animations from "css/animations";
+import { useData } from "@/lib/api/api";
+import * as branchesFragments from "@/lib/api/branches.fragments";
+
+/**
+ * Select a branch and handle login
+ * either the user is already logged in for that agency
+ * or the user needs to log in for that agency, so prompt login or open loanerform
+ * @param {obj} branch
+ * @param {obj} modal
+ */
+function handleOnSelect(branch, modal, context, updateLoanerInfo) {
+  // Selected branch belongs to one of the user's agencies where the user is logged in
+  const alreadyLoggedin = context.initial?.agencies?.find(
+    (agency) => agency.result?.[0].agencyId === branch.agencyId
+  );
+
+  // New selected branch has borrowercheck
+  const hasBorchk = branch.borrowerCheck;
+  // if selected branch has same origin as user agency
+  if (alreadyLoggedin && hasBorchk) {
+    // Set new branch without new log-in
+    updateLoanerInfo({ pickupBranch: branch.branchId });
+    // update context at previous modal
+    modal.prev();
+    return;
+  }
+  const callbackUID = modal?.stack?.find((m) => m.id === "order").uid;
+  if (branch?.borrowerCheck) {
+    modal.push("openAdgangsplatform", {
+      agencyId: branch.agencyId,
+      branchId: branch.branchId,
+      agencyName: branch.agencyName,
+      callbackUID: callbackUID, //we should always have callbackUID, but if we dont, order modal is not opened after login.
+    });
+    return;
+  } else {
+    modal.push("loanerform", {
+      branchId: branch.branchId,
+      changePickupBranch: true,
+    });
+  }
+}
+
+/**
+ * Special component responsible for loading order policy
+ * Will not render anything, but needs to be mounted
+ */
+function PolicyLoader({ branch, onLoad, pid, requireDigitalAccess }) {
```

Comment:
Ja, det er jo ærgerligt, at man ikke kan bruge hooks uden for components og hooks :sweat_smile: